### PR TITLE
EntityClockRoomController

### PR DESCRIPTION
**Pull request by sozud**
Starting to import work from https://github.com/Xeeynamo/sotn-decomp/pull/557
Credit to @sonicdcer

### CODE REVIEW

**Review by Xeeynamo — COMMENTED** [review]
I am close to submit a PR to de-duplicate data and functions. I hope you do not mind if we hold off from any MAR change until then :pray

**Comment by Xeeynamo** [comment]
It should be fine now. Note that `0x801803E4` is `g_eInitGeneric2` and `EntityClockRoomController` needs to go to `src/boss/mar/17FEC.c`

**Comment by sozud** [comment]
This is rebased and ready to be reviewed

File: src/boss/mar/17FEC.c [hunk 1]
```diff
@@ -1,7 +1,179 @@
...
+        }
+        break;
+
+    case 1:
+        if (status->timerFrames == 0) {
+            g_api.PlaySfx(0x7A9); // clock tick
+        }
+
+        // Update clock hands
+        self[minuteHand].ext.clockRoom.hand = status->timerMinutes * 0x3C;
```
Comment (sonicdcer) [inline src/boss/mar/17FEC.c:163]:
```suggestion
self[minuteHand].ext.clockRoom.hand = status->timerMinutes * 60;
```

Comment (sonicdcer) [reply src/boss/mar/17FEC.c:163]: I thought of this as a clock handle angle before, but Sozud recommended it to be changed to 60 in the original PR. Your choice.

**Comment by sozud** [comment]
I see that people are wasting time re-decompiling these so I'm just going to start merging https://decomp.me/scratch/6Bw71 (This is the same function as this one)

I will do a follow-up if there's any issues

**Pull request merged**